resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -156,8 +157,45 @@ public int compareTo(SortedKey<T> tMessageBodyKey)
       {
          return template;
       }
+
+       public T getObj() {
+           return obj;
+       }
    }
+   
+   protected static class ExtSortedKey<T> extends SortedKey<T>

```

Comment:
I do not see any reason for extending the SortedKey comparator. It seems that it provides no extra comparison logic as equal object comparison is already performed in the first step of SortedKey#compareTo.

Reply:
@sermojohn, given instances of
```
   @Provider
   @Priority(20)
   public static class ExceptionMapper1 implements ExceptionMapper<TestException> 
```
and
```
   @Provider
   @Priority(20)
   public static class ExceptionMapper2 implements ExceptionMapper<TestException> 
```
SortedKey.compareTo() will return 0 and ExtSortedKey.compareTo() will return -1. The latter will result in an insertion into sortedParamConverterProviders, and the former will not (because an "equal" element already exists). I.e., a ParamConverterProvider will get lost with SortedKey.

Reply:
You are totally right, I did not spot this!

Reply:
And that's what I spent yesterday doing. ;-)